jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -295,10 +303,130 @@
+     * @throws IllegalArgumentException if {@code graph} is not directed
+     */
+    public static <V, E> Graph<V, E> requireDirected(Graph<V, E> graph)
+    {
+        return requireDirected(graph, GRAPH_MUST_BE_DIRECTED);
+    }
+
+    /**
+     * Checks that the specified graph is undirected and throws a customized
+     * {@link IllegalArgumentException} if it is not. Also checks that the graph reference is not
+     * {@code null} and throws a {@link NullPointerException} if it is.
+     *
+     * @param graph the graph reference to check for being undirected and not null
+     * @param message detail message to be used in the event that an exception is thrown
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return {@code graph} if undirected and not {@code null}
+     * @throws NullPointerException if {@code graph} is {@code null}
+     * @throws IllegalArgumentException if {@code graph} is not undirected
+     */
+    public static <V, E> Graph<V, E> requireUndirected(Graph<V, E> graph, String message)
+    {
+        if (graph == null)
+            throw new NullPointerException(GRAPH_CANNOT_BE_NULL);
+        if (!graph.getType().isUndirected()) {
+            throw new IllegalArgumentException(message);
+        }
+        return graph;
+    }
+
+    /**
+     * Checks that the specified graph is undirected and throws an {@link IllegalArgumentException}
+     * if it is not. Also checks that the graph reference is not {@code null} and throws a
+     * {@link NullPointerException} if it is.
+     *
+     * @param graph the graph reference to check for being undirected and not null
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return {@code graph} if undirected and not {@code null}
+     * @throws NullPointerException if {@code graph} is {@code null}
+     * @throws IllegalArgumentException if {@code graph} is not undirected
+     */
+    public static <V, E> Graph<V, E> requireUndirected(Graph<V, E> graph)
+    {
+        return requireUndirected(graph, GRAPH_MUST_BE_UNDIRECTED);
+    }
+
+    /**
+     * Checks that the specified graph is directed or undirected and throws a customized
+     * {@link IllegalArgumentException} if it is not. Also checks that the graph reference is not
+     * {@code null} and throws a {@link NullPointerException} if it is.
+     *
+     * @param graph the graph reference to check for beeing directed or undirected and not null
+     * @param message detail message to be used in the event that an exception is thrown
+     * @param <V> the graph vertex type
+     * @param <E> the graph edge type
+     * @return {@code graph} if directed and not {@code null}
+     * @throws NullPointerException if {@code graph} is {@code null}
+     * @throws IllegalArgumentException if {@code graph} is mixed
+     */
+    public static <V, E> Graph<V, E> requireDirectedOrUndirected(Graph<V, E> graph, String message)
```

Comment:
I am using it a couple of times. As we slowly adapt several algorithms to accept both directed and undirected graphs, this will be very useful for constructor validation such as.

`this.graph = GraphTests.requireDirectedOrUndirected(graph)`

We could also make it private.